edx_django_utils/monitoring/tests/test_middleware.py

```diff
@@ -322,3 +324,59 @@ def get_mock_request(self, cookies_dict):
         for name, value in cookies_dict.items():
             factory.cookies[name] = value
         return factory.request()
+
+
+@ddt.ddt
+class FrontendMonitoringMiddlewareTestCase(TestCase):
+    """
+    Tests for FrontendMonitoringMiddleware.
+    """
+    def setUp(self):
+        super().setUp()
+        self.script = "<script>test script</script>"
+
+    @patch("edx_django_utils.monitoring.internal.middleware.FrontendMonitoringMiddleware.inject_script")
+    def test_frontend_middleware_without_setting_variable(self, mock_inject_script):
+        """
+        Test that middleware behaves correctly when setting variable is not defined.
+        """
+        response_html = '<html><head></head><body></body><html>'
+        middleware = FrontendMonitoringMiddleware(lambda r: HttpResponse(response_html, content_type='text/html'))
+        response = middleware(HttpRequest())
+        # Assert that the response content remains unchanged if settings not defined
+        assert response.content == response_html.encode()
```

Comment:
[nit, optional suggestion] If you were to rename `response_html` to `original_html` or similar, I think it would be easier to read the intent of lines like these.

Reply:
Done.